services/core-api/app/api/mines/documents/mine_document_search_util.py

```diff
@@ -0,0 +1,66 @@
+from app.api.projects.project_decision_package.models.project_decision_package import ProjectDecisionPackage
+from app.api.projects.project_decision_package.models.project_decision_package_document_xref import ProjectDecisionPackageDocumentXref
+from app.api.projects.project_summary.models.project_summary_document_xref import ProjectSummaryDocumentXref
+from app.api.projects.project_summary.models.project_summary import ProjectSummary
+from app.api.projects.information_requirements_table.models.information_requirements_table import InformationRequirementsTable
+from app.api.projects.information_requirements_table.models.information_requirements_table_document_xref import InformationRequirementsTableDocumentXref
+from app.api.mines.documents.models.mine_document import MineDocument
+from sqlalchemy import or_
+from app.extensions import db, api
+
+
+class MineDocumentSearchUtil():
+    """
+    Utility functions for searching mine documents.
+    This is split out from the MineDocument itself as to prevent circular dependencies
+    """
+    @classmethod
+    def filter_by(cls, mine_guid, project_guid=None, major_mine_application_guid=None, project_summary_guid=None, project_decision_package_guid=None, irt_id=None, is_archived=False):
+        """
+        Find Mine Documents by the given project related params
+        Returns a list of MineDocuments
+        """
+        qy = db.session.query(MineDocument).filter_by(mine_guid=mine_guid)
+
+        if is_archived is not None:
+            qy = qy.filter_by(is_archived=is_archived)
+
+        if project_guid is not None:
+            # A Project consists of the following loosely related entities (hence the many joins)
+            # - Major Mine Applications
+            # - Project Summary
+            # - Project Decision Package
+            # - Information Requirements Table
+            qy.join(MajorMineApplicationDocumentXref)\
+                .join(MajorMineApplication)\
+                .join(ProjectSummaryDocumentXref)\
+                .join(ProjectSummary)\
+                .join(ProjectDecisionPackageDocumentXref)\
+                .join(ProjectDecisionPackage)\
+                .join(InformationRequirementsTableDocumentXref)\
+                .join(InformationRequirementsTable)\
+                .filter(or_(
+                    MajorMineApplication.project_guid == project_guid,
+                    ProjectSummary.project_guid == project_guid,
+                    ProjectDecisionPackage.project_guid == project_guid,
+                    InformationRequirementsTable.project_guid == project_guid,
+                ))
+
+        if major_mine_application_guid is not None:
+            qy = qy.join(MajorMineApplicationDocumentXref)\
+                .join(MajorMineApplication)\
+                .filter(MajorMineApplication.major_mine_application_guid == major_mine_application_guid)
```

Comment:
I was wondering if there is need to also add deleted_ind == False as part of the filtering criteria.

Reply:
Good point! As far as I can tell we're actually not using the soft deletion functionality for the project related documents. When "deleting" them, we actually hard delete the Xref entry so they won't be included in the search. 
However definitely a good idea to add in the filter regardless for future proofing - will update the PR.